lib/gym/generators/package_command_generator_xcode7.rb

```diff
@@ -62,16 +69,86 @@ def config_path
         return Gym.cache[:config_path]
       end
 
+      # The path to the manifest plist file
+      def manifest_path
+        Gym.cache[:manifest_path] ||= File.join(temporary_output_path, "manifest.plist")
+      end
+
+      # The path to the app-thinning plist file
+      def app_thinning_path
+        Gym.cache[:app_thinning] ||= File.join(temporary_output_path, "app-thinning.plist")
+      end
+
+      # The path to the App Thinning Size Report file
+      def app_thinning_size_report_path
+        Gym.cache[:app_thinning_size_report] ||= File.join(temporary_output_path, "App Thinning Size Report.txt")
+      end
+
+      # The path to the Apps folder
+      def apps_path
+        Gym.cache[:apps_path] ||= File.join(temporary_output_path, "Apps")
+      end
+
       private
 
+      def normalize_export_options(hash)
+        # Normalize some values
+        hash[:onDemandResourcesAssetPacksBaseURL] = URI.escape(hash[:onDemandResourcesAssetPacksBaseURL]) if hash[:onDemandResourcesAssetPacksBaseURL]
+        hash[:manifest][:appURL] = URI.escape(hash[:manifest][:appURL]) if hash[:manifest][:appURL]
+        hash[:manifest][:displayImageURL] = URI.escape(hash[:manifest][:displayImageURL]) if hash[:manifest][:displayImageURL]
+        hash[:manifest][:fullSizeImageURL] = URI.escape(hash[:manifest][:fullSizeImageURL]) if hash[:manifest][:fullSizeImageURL]
+        hash[:manifest][:assetPackManifestURL] = URI.escape(hash[:manifest][:assetPackManifestURL]) if hash[:manifest][:assetPackManifestURL]
+        hash
+      end
+
+      def keys_to_symbols(hash)
+        # Convert keys to symbols
+        hash = hash.each_with_object({}) do |(k, v), memo|
+          memo[k.to_sym] = v
+          memo
+        end
+        hash
+      end
+
+      def read_export_options
+        # Reads export options
+        if Gym.config[:export_options]
+          if Gym.config[:export_options].kind_of?(Hash)
+            # Reads options from hash
+            hash = normalize_export_options(Gym.config[:export_options])
+          else
+            # Reads optoins from file
+            hash = Plist.parse_xml(Gym.config[:export_options])
+            # Convert keys to symbols
+            hash = keys_to_symbols(hash)
+          end
+
+          # Saves configuration for later use
+          Gym.config[:export_method] ||= hash[:method]
+          Gym.config[:include_symbols] = hash[:uploadSymbols] if Gym.config[:include_symbols].nil?
+          Gym.config[:include_bitcode] = hash[:uploadBitcode] if Gym.config[:include_bitcode].nil?
+          Gym.config[:export_team_id] ||= hash[:teamID]
+        # Generate export options plist
```

Comment:
I was make several refactoring. Probably I overlooked this comment. It should be removed.